cookbook/security/voters.rst

```diff
@@ -190,61 +248,128 @@ and tag it with ``security.voter``:
+
+        private $decisionManager;
+
+        public function __construct(AccessDecisionManagerInterface $decisionManager)
         {
-            // get a Post instance
-            $post = ...;
+            $this->decisionManager = $decisionManager;
+        }
 
-            // keep in mind that this will call all registered security voters
-            $this->denyAccessUnlessGranted('view', $post, 'Unauthorized access!');
+        protected function voteOnAttribute($attribute, $subject, TokenInterface $token)
+        {
+            // ...
+
+            // ROLE_SUPER_ADMIN can do anything! The power!
+            if ($this->decisionManager->decide($token, array('ROLE_SUPER_ADMIN'))) {
+                return true;
+            }
 
-            return new Response('<h1>'.$post->getName().'</h1>');
+            // ... all the normal voter logic
         }
     }
 
-.. versionadded:: 2.6
-    The ``denyAccessUnlessGranted()`` method was introduced in Symfony 2.6.
-    Prior to Symfony 2.6, you had to call the ``isGranted()`` method of the
-    ``security.context`` service and throw the exception yourself.
+Next, update ``services.yml`` to inject the ``security.access.decision_manager``
+service:
+
+.. configuration-block::
+
+    .. code-block:: yaml
+
+        # app/config/services.yml
+        services:
+            app.post_voter:
+                class: AppBundle\Security\PostVoter
+                arguments: ['@security.access.decision_manager']
+                tags:
+                    - { name: security.voter }
+
+    .. code-block:: xml
+
+        <!-- app/config/services.xml -->
+        <?xml version="1.0" encoding="UTF-8" ?>
+        <container xmlns="http://symfony.com/schema/dic/services"
+            xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+            xsi:schemaLocation="http://symfony.com/schema/dic/services
+                http://symfony.com/schema/dic/services/services-1.0.xsd">
+
+            <services>
+                <service id="app.post_voter"
+                    class="AppBundle\Security\Authorization\Voter\PostVoter"
+                    public="false"
+                >
+                    <argument type="service" id="security.access.decision_manager"/>
+
+                    <tag name="security.voter" />
+                </service>
+            </services>
+        </container>
+
+    .. code-block:: php
+
+        // app/config/services.php
+        use Symfony\Component\DependencyInjection\Definition;
+        use Symfony\Component\DependencyInjection\Reference;
+
+        $container->register('app.post_voter', 'AppBundle\Security\Authorization\Voter\PostVoter')
+            ->addArgument(new Reference('security.access.decision_manager'))
+            ->setPublic(false)
+            ->addTag('security.voter')
+        ;
+
+That's it! Calling ``decide()`` on the ``AccessDecisionManager`` is essentially
+the same as calling ``isGranted()`` on the normal ``security.authorization_checker``
```

Comment:
fair enough :) - I kept part of it, but removed `security.authorization_checker`. I want to highlight that `decide()` == `isGranted()`, so you understand what the heck this `decide()` thing is doing.